### ISSUE TITLE
Nested subquery forgets table

### ISSUE THREAD

**Issue by FaFre** [simolus3/moor#2010]
The code generator gives me following output:

```
Unknown table or view: p
ph.feature_id = p.feature_id
```

This is my boiled down query:
```sql
testQuery:
  SELECT
    (
      SELECT 
        true
      FROM (
        SELECT
          'test' AS name
        FROM
          place_hierarchy ph
        where
          ph.feature_id = p.feature_id
      ) second_select
    ) first_select
  FROM place_name matches 
  INNER JOIN place p ON p.feature_id = matches.feature_id; 
```
I stripped everything unnecessary out of the query, it doesn't make sense anymore, but I validated that SQLite is still processing it. From what I have tested, I can almost say for sure this is caused by the nested subquery. I'm currently on up to date develop.

**Comment by simolus3** [comment]
Very nice catch! Every time I think I have SQL's scoping rules figured out another peculiar edge case pops up :D

From a quick investigation, I think this because we hide all "outer" tables for nested subqueries appearing in a `FROM` clause. As sqlite3 does not support correlated queries, something like `SELECT * FROM foo AS outer, (SELECT * FROM bar WHERE x = outer.y)` is forbidden (as `outer` is not visible in the `FROM` clause).

However, the current logic is to just hide all outer tables for the inside of a `FROM` clause. It seems like we should just hide the ones from the immediately enclosing scope. As the broken reference appears via a chain from root -> subquery expression -> subquery in from, it should still be visible.

**Comment by simolus3** [comment]
Fixed in 33c13e2422a9337d06d0435ea7d614897b822ad4.
